pkg/apis/appmesh/v1beta1/types.go

```diff
@@ -453,6 +464,71 @@ type CloudMapServiceStatus struct {
 	NamespaceID *string `json:"namespaceId,omitempty"`
 }
 
+// General TLS Types
+
+type TlsValidationContext struct {
+	Trust TlsValidationContextTrust `json:"trust"`
+}
+
+type TlsValidationContextTrust struct {
+	// +optional
+	Acm *TlsValidationContextAcmTrust `json:"acm,omitempty"`
+	// +optional
+	File *TlsValidationContextFileTrust `json:"file,omitempty"`
+}
+
+type TlsValidationContextAcmTrust struct {
+	CertificateAuthorityArns []*string `json:"certificateAuthorityArns"`
```

Comment:
is there a reason for []*string? `[]string` is better to me, normally we only use *string to represent optional semantic. 

BTW, I know aws-sdk-go is a hell filled with pointers(and they are fixing it in aws-sdk-go-v2)

Reply:
I don't have a good reason, I think I was just going of some existing fields. There was an earlier discussion where I settled on exactly the point you're making, just didn't think to apply it to the slices.

Reply:
> BTW, I know aws-sdk-go is a hell filled with pointers

And it seems they have conversion helpers for everything because of it.